src/package-resolver.js

```diff
@@ -15,6 +15,12 @@ import WorkspaceLayout from './workspace-layout.js';
 const invariant = require('invariant');
 const semver = require('semver');
 
+export type ResolverOptions = {
```

Comment:
Can you make it an exact type using:

```
export type ResolverOptions = {|
  …
|}
```

that way you cannot add fields to the object that aren't specified here. In general, I recommend all object type definitions to use this unless they have a good reason not to.